CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+cmake_minimum_required(VERSION 3.12)
+
+set(CMAKE_DISABLE_SOURCE_CHANGES ON)
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
+
+project(flare LANGUAGES C CXX)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules/)
+
+include(ExternalProject)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+set(FLARE_PYTHON_VERSION "" CACHE STRING "Python version to use for compiling modules")
+
+if(NOT CMAKE_BUILD_TYPE)
+  message(STATUS "[flare] CMAKE_BUILD_TYPE not specified, setting it to "
+                 "Release. Use `-DCMAKE_BUILD_TYPE=...` to overwrite.")
+  set(CMAKE_BUILD_TYPE Release)
+endif()
+
+
+#
+# Dependencies
+#
+
+# googletest
+################################################################################
+
+# Download and unpack googletest at configure time
```

Comment:
I think this can be handled by externalProject, I think.

Or
```
FetchContent_Declare(googletest
    GIT_REPOSITORY https://github.com/google/googletest.git
    SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/googletest
    UPDATE_COMMAND ""
)
FetchContent_MakeAvailable(googletest)
```

Reply:
I just followed the wise counsel of Google here; see in particular the section "Incorporating into an existing cmake project":
https://github.com/google/googletest/blob/master/googletest/README.md

Reply:
CMake 3.11+ has FetchContent, which works like externalproject+add_subdirectory.

Reply:
Okay, I switched to FetchContent in 68e6443

Reply:
Looks like I'll need to change the minimum required cmake version to 3.14.5:
https://github.com/EMsoft-org/EMsoft/issues/93
(I had trouble compiling with 3.12.1 on Odyssey)